coucharchive

```diff
@@ -253,26 +253,43 @@ class ReplicationControl(object):
         self.ideal_speed = 0
         self.current_avge_speed = 0
 
-        self._last_successes = multiprocessing.Manager().list()
-        self._last_errors = multiprocessing.Manager().list()
+        self._last_successes_reported = multiprocessing.Queue()
+        self._last_successes = []
+        self._last_errors_reported = multiprocessing.Queue()
+        self._last_errors = []
 
     def report_success(self):
-        self._last_successes.append((int(time.time()),
-                                     self.running_replications.value))
+        self._last_successes_reported.put((int(time.time()),
+                                           self.running_replications.value))
 
     def report_error(self):
-        self._last_errors.append((int(time.time()),
-                                  self.running_replications.value))
+        self._last_errors_reported.put((int(time.time()),
+                                        self.running_replications.value))
 
-    def _drop_old_events(self):
+    def _clean_and_read_events(self):
         ten_min_ago = time.time() - 10 * 60
-        self._last_successes[:] = [e for e in self._last_successes
-                                   if e[0] > ten_min_ago]
-        self._last_errors[:] = [e for e in self._last_errors
+        self._last_successes = [e for e in self._last_successes
                                 if e[0] > ten_min_ago]
+        self._last_errors = [e for e in self._last_errors
+                             if e[0] > ten_min_ago]
```

Comment:
It would be more logical to clean *at the end* (and rename the function consequently), no? Because we can imagine that old events (> 10 min) arrive in `self._last_successes_reported` (I agree it can hardly happen in real life, but it seems more natural to write it like that!)

Reply:
It will not change much, and will be more clear and intuitive your are right.
Here is the diff!
```diff
--- a/coucharchive
+++ b/coucharchive
@@ -267,12 +267,6 @@ class ReplicationControl(object):
                                         self.running_replications.value))
 
     def _clean_and_read_events(self):
-        ten_min_ago = time.time() - 10 * 60
-        self._last_successes = [e for e in self._last_successes
-                                if e[0] > ten_min_ago]
-        self._last_errors = [e for e in self._last_errors
-                             if e[0] > ten_min_ago]
-
         reported = []
         while True:
             try:
@@ -288,6 +282,12 @@ class ReplicationControl(object):
                 break
         self._last_errors += reported
 
+        ten_min_ago = time.time() - 10 * 60
+        self._last_successes = [e for e in self._last_successes
+                                if e[0] > ten_min_ago]
+        self._last_errors = [e for e in self._last_errors
+                             if e[0] > ten_min_ago]
+
     def recent_errors(self):
         self._clean_and_read_events()
         return len(self._last_errors)